src/components/Header/Header.module.scss

```
@@ -1,18 +1,39 @@
-
 .header {
   position: sticky;
   top: 0;
-  background: $button-primary-text;
+  background: rgba($button-primary-text, 0.8);
+  backdrop-filter: blur(10px);
+  -webkit-backdrop-filter: blur(10px);
   z-index: 100;
+  transition: all 0.3s ease 0s;
+  
+
+  html[data-theme="dark"] & {
+    background: rgba($primary-color, 0.8);
+    backdrop-filter: blur(10px);
+    -webkit-backdrop-filter: blur(10px);
+    transition: all 0.3s ease 0s;
```

как вариант, можно добавить миксин animate, чтобы передавать только свойство, которое нужно анимировать (и в целом лучше указывать определенные свойства, а не all, если возможно) и, возможно, длительность, если в каком-то месте ее нужно будет изменить
```suggestion
    @include animate(background);
```

---

Учту на будущее)